pkg/scorecard/json.go

```diff
@@ -160,10 +159,10 @@ func (r *Result) AsJSON2(writer io.Writer, checkDocs docs.Doc, opt *AsJSON2Resul
 	for _, checkResult := range r.Checks {
 		doc, e := checkDocs.GetCheck(checkResult.Name)
 		if e != nil {
-			return sce.WithMessage(sce.ErrScorecardInternal, fmt.Sprintf("GetCheck: %s: %v", checkResult.Name, e))
+			return out, fmt.Errorf("GetCheck: %s: %w", checkResult.Name, e)
 		}
 		if doc == nil {
-			return sce.WithMessage(sce.ErrScorecardInternal, fmt.Sprintf("GetCheck: %s: %v", checkResult.Name, errNoDoc))
+			return out, fmt.Errorf("GetCheck: %s: %w", checkResult.Name, errNoDoc)
```

Comment:
changing the error return like this is a breaking change here if any library callers are checking for `ErrScorecardInternal`.

Reply:
Oops, forgot you made this a helper, it's still a `ErrScorecardInternal` for `AsJSON2`, so no breaking change here.